python/tests/kernel/test_kernel_complex.py

```diff
@@ -0,0 +1,536 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import sys
+
+import pytest
+import numpy as np
+
+import cudaq
+
+## [PYTHON_VERSION_FIX]
+skipIfPythonLessThan39 = pytest.mark.skipif(
+    sys.version_info < (3, 9),
+    reason="built-in collection types such as `list` not supported")
+
+
+def is_close(expected, actual):
+    return np.isclose(expected, actual, atol=1e-6)
+
+
+def test_complex_params():
+    """Test that we can pass complex lists to kernel functions."""
+
+    # Pass a list of complex as a parameter
+    c = [.70710678 + 1j, 2j, 0., 0.70710678]
+
+    @cudaq.kernel
+    def complex_vec_param(vec: list[complex], i: int) -> complex:
+        return vec[i]
+
+    # Returning complex is not supported yet
```

Comment:
any reason not to support this here. seems like it'd be a simple update to py_alt_launch_kernel where you already updated for F32

Reply:
Will add complex return types in a separate PR

Reply:
I think a train of PRs, each one a smaller chunk of work that's easier to review, is a good idea. 👍